src/device.hpp

```diff
@@ -65,8 +65,10 @@ class VmuxDevice {
     /* vfio endpoint, may be null for some devices */
     std::shared_ptr<VfioConsumer> vfioc;
 
+    std::shared_ptr<VfioUserServer> vfuServer;
```

Comment:
with that we can now actually remove the ugly dynamic_pointer_casts from vfio-server